docs/linking/README.md

```diff
@@ -1,3 +1,255 @@
-Linking via OAuth2 and FIDO
+<h1>Contents</h1>
 
-BCP on Native app OAuth - https://tools.ietf.org/html/rfc8252
\ No newline at end of file
+<!-- TOC -->
+
+- [1. Linking](#1-linking)
+    - [1.1. Pre-linking](#11-pre-linking)
+    - [1.2. Discovery](#12-discovery)
+    - [1.3. Request consent](#13-request-consent)
+        - [1.3.1. Web](#131-web)
+        - [1.3.2. OTP / SMS](#132-otp--sms)
+    - [1.4. Authentication](#14-authentication)
+        - [1.4.1. Web](#141-web)
+        - [1.4.2. OTP](#142-otp)
+    - [1.5. Grant consent](#15-grant-consent)
+    - [1.6. Credential registration](#16-credential-registration)
+        - [1.6.1. Requesting a challenge](#161-requesting-a-challenge)
+        - [1.6.2. Finalizing the credential](#162-finalizing-the-credential)
+- [2. Unlinking](#2-unlinking)
+- [3. Third-party credential registration](#3-third-party-credential-registration)
+    - [3.1. Authentication](#31-authentication)
+    - [3.2. Credential registration](#32-credential-registration)
+
+<!-- /TOC -->
+
+# 1. Linking
+
+The goal of the linking process is to explain how users establish trust between
+all three interested parties:
+
+1. User
+2. DFSP where User has an account
+3. PISP that User wants to rely on to initiate payments
+
+*NOTE: For now, we're focusing on the FIDO / Web authentication channel.*
+
+Linking is broken down into several separate phases:
+
+1. **Pre-linking**
+   In this phase, a PISP asks what DFSPs are available to link with.
+2. **Request consent**
+   In this phase, a PISP attempts to establish trust between the 3 parties.
+3. **Authentication**
+   In this phase, a User proves their identity to their DFSP.
+4. **Grant consent**
+   In this phase, a PISP proves to the DFSP that the User and PISP have
+   established trust and, as a result, the DFSP confirms that mutual trust
+   exists between the 3 parties.
+5. **Credential registration**
+   In this phase, a User establishes the credential they'll use to consent to
+   future transfers from the DFSP and initiated by the PISP.
+
+## 1.1. Pre-linking
+
+In this phase, a PISP Server needs to know what DFSPs are available to link
+with. This is *unlikely* to be done on-demand (e.g., when a User clicks "link"
+in the PISP mobile App), and far more likely to be done periodically and cached
+by the PISP Server. The reason for this is simply that new DFSPs don't typically
+join the Mojaloop network all that frequently, so calling this multiple times on
+the same day (or even the same month) would likely yield the same results.
+
+The end-goal of this phase is for the PISP Server to have a final list of DFSPs
+available and any relevant metadata about those DFPSs that are necessary to
+begin the linking process.
+
+![Pre-linking](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/mojaloop/pisp/master/docs/linking/0-pre-linking.puml)
```

Comment:
Getting an error for "Non image content"  on such diagrams in the docs/inking/README.md file

Reply:
Yes -- these will only render correctly once the file is merged (and therefore living in mojaloop/pisp/master -- currently the file lives in jgeewax/pips/linking-docs)

Reply:
In other words, if you try http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/jgeewax/pisp/linking-docs/docs/linking/0-pre-linking.puml (note that I've just done `s~mojaloop/pisp/master~jgeewax/pisp/linking-docs~`) it should work just fine. Once this is merged, the current URL will become correct.

Reply:
Thanks for the clarification